src/features/utils/instrument-base.js

```diff
@@ -0,0 +1,37 @@
+import { registerDrain } from "../../common/drain/drain"
+import { FeatureBase } from "./feature-base"
+import { onWindowLoad } from '../../common/window/load'
+import { isWorkerScope } from "../../common/util/global-scope"
+import { FEATURE_NAMES } from "../../loaders/features/features"
+
+export class InstrumentBase extends FeatureBase {
+  constructor(agentIdentifier, aggregator, featureName, auto = true) {
+    super(agentIdentifier, aggregator, featureName)
+    this.completed = new Promise((resolve, reject) => {
+      this.resolve = resolve
+      this.reject = reject
+    })
+    this.hasAggregator = false
+    this.auto = auto
+
+    if (auto && featureName !== FEATURE_NAMES.sessionTrace) registerDrain(agentIdentifier, featureName)
```

Comment:
Since you have logic in drain to let things pass through the gate if they're a) tests, or b) "loaders that directly import [features'] aggregators",

1. the default for `auto` is false for pageaction (only at this time). Wouldn't this mean pageaction also fails to register itself and isn't harvest gated?
2. this seems to be the centralized (good) and only place `registerDrain` is called for all the features. That means `sessionTrace` would never register itself either.
3. given the exceptions to the drain-gating logic, can we be confident that in no other scenario, `auto` would be false even if instrumentation files were imported? Want to make sure gating is the **default** behavior in all other scenarios that's not (a) or (b).

Reply:
the auto in the configs is a remnant from when we were trying to mash the MFE loader and regular loader into one thing and can be removed.  The "auto" used here is just a constructor argument used by tests to disable "auto" importing the agg which has side-effects that are obstructive in some tests.

Reply:
https://issues.newrelic.com/browse/NEWRELIC-6480